### PR TITLE
Update msg

### LINKED ISSUE #5: Is there a way to log progress?
This is a fantastic package and thanks so much for making it available!

Is there a way to log progress? For instance, I'd love to give the status of a predicate.

I'm using this to way for a Celery group to finish and I would find it incredibly useful to do something like this following:

    wait(lambda: res.ready(),
         waiting_for="tasks ({0}) to finish".format(','.join(task.name for task in tasks)),
         logger=logger,
         info="{0} tasks finished".format(res.completed_count()),
         timeout_seconds=timeout)

so it logs the info every `sleep_seconds` seconds.

Is there any way to do this currently? I didn't see any examples that covered this.

### PR DESCRIPTION
**Pull request by erip**
Add option to pass lambda for logging purposes. Usage demonstrated in README. Closes #5.

### CODE REVIEW

**Comment by erip** [comment]
There seems to be a funny issue with Travis... it isn't handling the tests very well... timing out, etc. Hmmm.

File: README.rst [hunk 1]
```diff
@@ -80,3 +80,15 @@ If your predicate might raise certain exceptions you wish to ignore, you may use
...
  >>> wait(predicate, expected_exceptions=(ValueError, AttributeError))
  True
+
+If you'd like to maintain updates while waiting for a predicate to complete, you may use ``update_msg`` to pass a lambda to log the status. By default, this is a no-op.
+
+  >>> import logging
+  >>> from waiting import wait
+  >>> try:
+  ...    wait(lambda: False, timeout_seconds=5,               # Timeout after 5 seconds
+  ...          update_msg=lambda: logging.warn("Waiting...")) # Log "Waiting..." six times.
```
Comment (vmalloc) [inline README.rst]: update_msg is not a good name for "just a function that gets called on every poll loop".

I think "on_poll" is a much better name

File: waiting/__init__.py [hunk 2]
```diff
@@ -19,12 +19,15 @@ def wait(*args, **kwargs):
 
 
 def iterwait(predicate, timeout_seconds=None, sleep_seconds=1, result=None, waiting_for=None,
-             expected_exceptions=()):
+             on_poll=lambda: None, expected_exceptions=()):
```
Comment (vmalloc) [inline waiting/__init__.py]: One more nitpick: It's better to default `on_poll` to `None` than to a no-op lambda. It makes people wonder less when using `help()` or inspecting the arguments...

Comment (erip) [inline waiting/__init__.py]: Done.

**Comment by vmalloc** [comment]
Thanks!

**Comment by erip** [comment]
@vmalloc Thanks for the merge! I seem to have added an extra space to the README.rst, so it's rendering as a quote-block in Markdown. It might be easier for you to fix that. If not, I'd happily submit another PR. 😄

**Pull request merged**